### PR TITLE
Code of Conduct added in Footer

### PR DESCRIPTION
**Pull request by aSquare14**
# Description 
This PR is related to issue #902 Add code of conduct for allies/groups. 
Please write Code Of Conduct.

## More Details
Code of conduct that will appear and/or get linked to when groups are created or joined
Adding a link to CoC in website footer

## Corresponding Issue
 #902

### CODE REVIEW

File: app/views/pages/codeofconduct.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<% title t('navigation.codeofconduct') %>
+
+<div class="main_message">
+ Code Of Conduct
```
Comment (julianguyen) [inline app/views/pages/codeofconduct.html.erb]: So the CoC we use the [Contributor Covenant](https://www.contributor-covenant.org/). It's actually already translated to various languages (which is rad!). I think we should just link directly to those pages.

Here's our copy of it in the repo: https://github.com/ifmeorg/ifme/blob/master/code_of_conduct.md

File: config/locales/en.yml [hunk 2]
```diff
@@ -43,6 +43,7 @@ en:
    press: 'Press'
    faq: 'FAQ'
    resources: 'Resources'
+   codeofconduct: 'Code Of Conduct'
```
Comment (julianguyen) [inline config/locales/en.yml]: You would have to translate this to the other languages we support

File: config/locales/en.yml [hunk 3]
```diff
@@ -334,7 +335,7 @@ en:
     comment_text: "<p><i>%{comment}</i></p>"
     comment_link: "<p>To read it, %{link}!</p>"
     new_group_subject: "if me | %{user} created a group \"%{group}\""
-    new_group_body: "<p>%{user} created a group \"%{group}\":</p><p><i>%{description}</i></p><p>To learn more and join, %{link}!</p>"
+    new_group_body: "<p>%{user} created a group \"%{group}\":</p><p><i>%{description}</i></p><p>To learn more and join, %{link}!</p><p> To see our code of conduct %{link_to_coc} !</p>"
```
Comment (julianguyen) [inline config/locales/en.yml]: Same goes here.

File: config/routes.rb [hunk 4]
```diff
@@ -90,6 +90,7 @@
   match 'toggle_locale', to: 'pages#toggle_locale', via: :get
   match 'press', to: 'pages#press', via: :get
   match 'resources', to: 'pages#resources', via: :get
+  match 'codeofconduct', to: 'pages#codeofconduct', via: :get
```
Comment (julianguyen) [inline config/routes.rb]: I would rename this to `code_of_conduct`

**Comment by aSquare14** [comment]
@julianguyen  I am unable to understand why the CircleCi build is failing :(

**Comment by julianguyen** [comment]
@aSquare14 Can you try running `cd client && RAILS_ENV=test bundle exec rake react_on_rails:locale && yarn lint:setup && yarn lint:build && yarn build:test && yarn test` locally? This is the command that is being run on CircleCI and getting that error. Let's see if we can reproduce it locally.

**Comment by aSquare14** [comment]
@julianguyen  Yay. Tests are going green :)

File: app/helpers/notification_mailer_helper.rb [hunk 5]
```diff
@@ -52,7 +52,9 @@ def new_group_body(data)
       user: data['user'].to_s,
       group: data['group'].to_s,
       description: Group.where(id: data['group_id']).first.description,
-      link: link_to(link_name, group_url(data['group_id']))
+      link: link_to(link_name, group_url(data['group_id'])),
```
Comment (julianguyen) [inline app/helpers/notification_mailer_helper.rb]: I would update the tests to check for this

File: config/locales/en.yml [hunk 6]
```diff
@@ -334,7 +335,7 @@ en:
     comment_text: "<p><i>%{comment}</i></p>"
     comment_link: "<p>To read it, %{link}!</p>"
     new_group_subject: "if me | %{user} created a group \"%{group}\""
-    new_group_body: "<p>%{user} created a group \"%{group}\":</p><p><i>%{description}</i></p><p>To learn more and join, %{link}!</p>"
+    new_group_body: "<p>%{user} created a group \"%{group}\":</p><p><i>%{description}</i></p><p>To learn more and join, %{link}!</p><p>To see our code of conduct %{link_to_code_of_conduct}!</p>"
```
Comment (julianguyen) [inline config/locales/en.yml]: We should change the copy from`To see our code of conduct %{link_to_code_of_conduct}!` to `Please follow our %{link_to_code_of_conduct}!` where link_to_code_of_conduct is `link_to(t('navigation.code_of_conduct'), 'https://www.contributor-covenant.org/')`

**Comment by aSquare14** [comment]
@julianguyen Unable to figure out why tests are failing :(

**Comment by aSquare14** [comment]
@julianguyen  Tests passed ! Yay :D

**Pull request merged**